save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt

```diff
@@ -252,28 +254,29 @@ class LnkUserOrganizationController(
         lnkUserOrganizationService.getSuperOrganizationsWithRole((authentication.details as AuthenticationDetails).id)
     )
 
-    @GetMapping("/by-user/not-deleted")
+    @GetMapping("/by-user")
     @RequiresAuthorizationSourceHeader
     @PreAuthorize("permitAll()")
     @Operation(
         method = "GET",
         summary = "Get user's organizations.",
-        description = "Get not deleted organizations where user is a member, and his roles in those organizations.",
+        description = "Get organizations with status where user is a member, and his roles in those organizations.",
     )
     @ApiResponse(responseCode = "200", description = "Successfully fetched organization infos.")
-    @ApiResponse(responseCode = "404", description = "Could not find user with this id.")
+    @ApiResponse(responseCode = "404", description = "Could not find user with this id and status.")
     @Suppress("UnsafeCallOnNullableType")
-    fun getOrganizationWithRoles(
+    fun getOrganizationWithRolesAndStatus(
         authentication: Authentication,
+        @RequestParam status: OrganizationStatus?,
     ): Flux<OrganizationDto> = Mono.justOrEmpty(
         lnkUserOrganizationService.getUserById((authentication.details as AuthenticationDetails).id)
     )
         .switchIfEmptyToNotFound()
         .flatMapMany {
             Flux.fromIterable(lnkUserOrganizationService.getOrganizationsAndRolesByUser(it))
         }
-        .filter {
-            it.organization != null && it.organization?.status != OrganizationStatus.DELETED
+        .filter {lnkUserOrganization ->
+            lnkUserOrganization.organization != null && (status?.let { lnkUserOrganization.organization?.status == it } ?: true)
```

Comment:
why `lnkUserOrganization.organization` is nullable? @Cheshiriks as I can see you added `lnkUserOrganization`. Do we have use cases when we have a link between user and organization and organization is null?

Reply:
@Arrgentum can you please try to fix it in your PR?

Reply:
I'll take it